runtime/interpreter/encode.go

```diff
@@ -755,13 +771,13 @@ func (e *Encoder) prepareDictionaryValue(
 			if err != nil {
 				return nil, err
 			}
-			entries[key] = prepared
+			entries[i] = prepared
 		}
 	}
 
 	return cbor.Tag{
 		Number: cborTagDictionaryValue,
-		Content: cborMap{
+		Content: cborArray{
```

Comment:
Hah, nice that the slice initialization can stay the same, I didn't know this was possible 👍